react-hot-button-component/src/hot-button.jsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+// import reactDOM from 'react-dom/client';
+class HotBtn extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+    this.state = { clicks: 0 };
+  }
+
+  handleClick() {
+    const updateCount = this.state.clicks + 1;
+    this.setState({ clicks: updateCount });
+  }
+
+  render() {
+
+    if (this.state.clicks < 4) {
+      return <button className="btn-blue" onClick={this.handleClick}>Hot Button</button>;
+    } else if (this.state.clicks >= 3 && this.state.clicks < 6) {
+      return <button className="btn-purple" onClick={this.handleClick}>Hot Button</button>;
+    } else if (this.state.clicks >= 6 && this.state.clicks < 12) {
+      return <button className="btn-pink" onClick={this.handleClick}>Hot Button</button>;
+    } else if (this.state.clicks >= 12 && this.state.clicks < 15) {
+      return <button className="btn-orange" onClick={this.handleClick}>Hot Button</button>;
+    } else if (this.state.clicks >= 15 && this.state.clicks < 18) {
+      return <button className="btn-yellow" onClick={this.handleClick}>Hot Button</button>;
+    } else if (this.state.clicks >= 18 && this.state.clicks < 21) {
+      return <button className="btn-white" onClick={this.handleClick}>Hot Button</button>;
+    }
+  }
```

Comment:
This works but would be hard to maintain and update. A better solution would be something like:
```JSX
render() {
  let btnClass = 'btn-blue';

  if (this.state.clicks >= 3 && this.state.clicks < 6) {
    btnClass = "btn-purple";
  } else if (this.state.clicks >= 6 && this.state.clicks < 12) {
    btnClass = "btn-pink";
  } else if (this.state.clicks >= 12 && this.state.clicks < 15) {
    btnClass = "btn-orange";
  } else if (this.state.clicks >= 15 && this.state.clicks < 18) {
    btnClass = "btn-yellow";
  } else if (this.state.clicks >= 18 && this.state.clicks < 21) {
    btnClass = "btn-white";
  }

  return <button className={btnClass} onClick={this.handleClick}>Hot Button</button>;
}
```

Reply:
https://user-images.githubusercontent.com/99847558/198697236-1fd6c4f0-173f-485a-bd9c-9b27396d4d83.mov